### PR TITLE
Updates for tests from ipy2

### PR DESCRIPTION
**Pull request by slide**

### CODE REVIEW

**Comment by slide** [comment]
It is a merge of changes from the tests folder and below mostly

**Comment by slozier** [comment]
All these changes look good. I was just wondering since I know you did a bunch of renaming of tests to match the format test_xxx which I'm not seeing here.

**Comment by slide** [comment]
I think some of that was done already, are there specific files I can check?

**Comment by slozier** [comment]
All these are in ipy2 but not ipy3, I assume most are renames:
Tests/functions/test_signatures.py
Tests/hosting/editor_svcs/test_errorlistener.py
Tests/hosting/editor_svcs/test_tokencategorizer.py
Tests/interop/net/dynamic/test_dynamic_regressions.py
Tests/interop/net/versions/test_four_dot_zero.py
Tests/modules/io_related/test__bytesio.py
Tests/modules/io_related/test__fileio.py
Tests/modules/io_related/test_binascii.py
Tests/modules/io_related/test_cPickle.py
Tests/modules/io_related/test_cStringIO.py
Tests/modules/io_related/test_codecs.py
Tests/modules/io_related/test_copy_reg.py
Tests/modules/io_related/test_fd.py
Tests/modules/io_related/test_io_StringIO.py
Tests/modules/io_related/test_marshal.py
Tests/modules/io_related/test_re.py
Tests/modules/misc/test__random.py
Tests/modules/misc/test__sha256.py
Tests/modules/misc/test__sha512.py
Tests/modules/misc/test__warnings.py
Tests/modules/misc/test__weakref.py
Tests/modules/misc/test_ctypes.py
Tests/modules/misc/test_datetime.py
Tests/modules/misc/test_future.py
Tests/modules/misc/test_math.py
Tests/modules/misc/test_operator.py
Tests/modules/misc/test_system_namespaces.py
Tests/modules/misc/test_zlib.py
Tests/modules/network_related/test__socket.py
Tests/modules/network_related/test__ssl.py
Tests/modules/system_related/test__locale.py
Tests/modules/system_related/test_gc.py
Tests/modules/system_related/test_nt.py
Tests/modules/system_related/test_signal.py
Tests/modules/system_related/test_sys.py
Tests/modules/system_related/test_sys_getframe.py
Tests/modules/system_related/test_thread.py
Tests/modules/system_related/test_time.py
Tests/modules/type_related/test__struct.py
Tests/modules/type_related/test_array.py
Tests/modules/type_related/test_types.py

**Comment by slide** [comment]
This should handle IronPython/ironpython2#259, IronPython/ironpython2#255, IronPython/ironpython2#252

File: Tests/ImportTestDir/pkg_a/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (slozier) [inline Tests/ImportTestDir/pkg_a/__init__.py]: I think `ImportTestDir` folder is a test artifact of `test_importpkg` and probably shouldn't be in git.

File: Tests/interop/net/test_loadorder.py [hunk 2]
```diff
@@ -1,124 +0,0 @@
-#####################################################################################
```
Comment (slozier) [inline Tests/interop/net/test_loadorder.py]: I probably asked this in the original PR, but why is this file getting removed? I'm not seeing a replacement.

**Comment by slozier** [comment]
Also covers: https://github.com/IronLanguages/ironpython2/pull/224 https://github.com/IronLanguages/ironpython2/pull/228

File: Tests/interop/net/test_loadorder.py (hunk 2 above)
Comment (slide) [reply Tests/interop/net/test_loadorder.py]: Here's the comment for when I did it in ipy2

https://github.com/IronLanguages/ironpython2/pull/2#discussion_r119403155

**Comment by slide** [comment]
Any other feedback on this?

**Pull request merged**